django/contrib/auth/management/__init__.py

```diff
@@ -110,82 +110,74 @@ def create_permissions(
             print("Adding permission '%s'" % perm)
 
 
-class RenamePermission(migrations.RunPython):
-    def __init__(self, app_label, old_model, new_model):
-        self.app_label = app_label
-        self.old_model = old_model
-        self.new_model = new_model
-        super(RenamePermission, self).__init__(
-            self.rename_forward, self.rename_backward
-        )
+def rename_permissions_after_model_rename(
+    app_config, verbosity=0, plan=None, using=None, apps=global_apps, **kwargs
+):
+    # This handler is connected to the global post_migrate signal, which is
+    # emitted for *all* apps — including test configurations where
+    # django.contrib.auth is NOT installed.
+    # Therefore, we must guard against 'auth' not being present, otherwise
+    # apps.get_model("auth", "Permission") will raise LookupError and break
+    # unrelated test suites (especially under parallel test execution).
+    try:
+        Permission = apps.get_model("auth", "Permission")
+    except LookupError:
+        return
 
-    def _rename(self, apps, schema_editor, old_model, new_model):
-        ContentType = apps.get_model("contenttypes", "ContentType")
-        # Use the live Permission model instead of the frozen one, since frozen
-        # models do not retain foreign key constraints.
-        from django.contrib.auth.models import Permission
+    db = using or router.db_for_write(Permission)
+
+    if not router.allow_migrate_model(db, Permission):
+        return
```

Comment:
For my own knowledge, why the pattern change here versus `create_permissions()`?